### PR TITLE
[Skin] Add video scan type info to player process info window

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

This add scan type info (progressive / interlaced) to the resolution value in the `Player` page of player process info window

## Motivation and context

Seen the new recent restyling to this window @Portisch asked me if it was possible to integrate this new info already available on CoreElec to make it official on Kodi, therefore this code is not completely fruit of my bag

## How has this been tested?

With an interlaced and progressive video and with ISAdaptive see screenshots

## Screenshots (if appropriate):
immagine
immagine

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -971,6 +971,14 @@ const infomap player_times[] =   {{ "seektime",         PLAYER_SEEKTIME },
 ///     @skinning_v17 **[New Infolabel]** \link Player_Process_videoheight `Player.Process(videoheight)`\endlink
 ///     <p>
 ///   }
+///   \table_row3{   <b>`Player.Process(videoscantype)`</b>,
+///                  \anchor Player_Process_videoscantype
+///                  _string_,
+///     @return The scan type (progressive / interlaced) of the currently playing video.
```
Comment (enen92) [inline xbmc/GUIInfoManager.cpp:977]: You seem to return "p" and "i" in the infolabel implementation and not progressive/interlaced.

```suggestion
///     @return The scan type identifier of the currently playing video **p** (for progressive) or **i** (for interlaced).
```

File: xbmc/cores/DataCacheCore.cpp [hunk 2]
```diff
@@ -188,6 +188,20 @@ float CDataCacheCore::GetVideoDAR()
   return m_playerVideoInfo.dar;
 }
 
+void CDataCacheCore::SetVideoInterlaced(bool isInterlaced)
+{
+  CSingleLock lock(m_videoPlayerSection);
+
```
Comment (enen92) [inline xbmc/cores/DataCacheCore.cpp:194]: no need for the empty line as the implementation is really small

File: xbmc/cores/DataCacheCore.cpp [hunk 3]
```diff
@@ -188,6 +188,20 @@ float CDataCacheCore::GetVideoDAR()
...
+  m_playerVideoInfo.isInterlaced = isInterlaced;
+}
+
+bool CDataCacheCore::IsVideoInterlaced()
+{
+  CSingleLock lock(m_videoPlayerSection);
+
```
Comment (enen92) [inline xbmc/cores/DataCacheCore.cpp:201]: no need for the empty line as the implementation is really small

File: xbmc/GUIInfoManager.cpp (hunk 1 above)
Comment (CastagnaIT) [reply xbmc/GUIInfoManager.cpp:977]: you are right so it is more understandable

File: xbmc/cores/DataCacheCore.h [hunk 4]
```diff
@@ -154,6 +156,7 @@ class CDataCacheCore
     int height;
     float fps;
     float dar;
+    bool isInterlaced;
```
Comment (enen92) [inline xbmc/cores/DataCacheCore.h:159]: `m_` according to the code guidelines

File: xbmc/cores/DataCacheCore.h [hunk 5]
```diff
@@ -45,6 +45,8 @@ class CDataCacheCore
   float GetVideoFps();
   void SetVideoDAR(float dar);
   float GetVideoDAR();
+  void SetVideoInterlaced(bool isInterlaced);
```
Comment (enen92) [inline xbmc/cores/DataCacheCore.h:48]: missing doxygen signatures for new added methods

File: xbmc/cores/DataCacheCore.h (hunk 4 above)
Comment (CastagnaIT) [reply xbmc/cores/DataCacheCore.h:159]: in a `struct` ? it is a new rule?

Comment (enen92) [reply xbmc/cores/DataCacheCore.h:159]: https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#member-variables

Code guidelines mention member variables and don't specify a scope. So my interpretation is that struct member variables are also covered by this. I won't force it though.

**Review by ronie — APPROVED** [review]
skin changes look good to me

File: xbmc/cores/DataCacheCore.h (hunk 4 above)
Comment (CastagnaIT) [reply xbmc/cores/DataCacheCore.h]: ok changed

**Review by enen92 — APPROVED** [review]

**Review by garbear — APPROVED** [review]
I reviewed the whole thing and didn't find any issues. Good work.

**Comment by enen92** [comment]
Updated forum in: https://forum.kodi.tv/showthread.php?tid=363553&pid=3079232#pid3079232

**Pull request merged**